### PR TITLE
Support scientific notation

### PR DESCRIPTION
**Pull request by logansnow17**

### CODE REVIEW

File: test/integration/parsing/general.js [hunk 1]
```diff
@@ -40,4 +40,36 @@ describe('.parse() general', () => {
...
+  it('should parse scientific notation within complex formula respecting order of operations 2', () => {
+    expect(parser.parse('(2*1e2)^3')).toMatchObject({error: null, result: 8000000});
+  });
+
+  it('should not parse with non numeric tokens', () => {
+    expect(parser.parse('(2)e(4)')).toMatchObject({error: "#ERROR!", result: null});
+  });
+
+  it('should parse with decimal tokens 1', () => {
+    expect(parser.parse('0.2e2)')).toMatchObject({error: null, result: 20});
```
Comment (logansnow17) [inline test/integration/parsing/general.js]: This and the following test are failing

File: src/grammar-parser/grammar-parser.jison [hunk 2]
```diff
@@ -222,6 +223,9 @@ number
   | number '%' {
       $$ = $1 * 0.01;
     }
+  | NUMBER 'e' NUMBER {
+      $$ = yy.evaluateByOperator('*', [$1, yy.evaluateByOperator('^', [10, $3])])
+    }
```
Comment (jhirshman) [inline src/grammar-parser/grammar-parser.jison:234]: decimal_number: 
(the first three cases)

number:
- decimal_number
- decimal_number e decimal_number
- decimal_number %

File: test/integration/parsing/general.js [hunk 3]
```diff
@@ -40,4 +40,36 @@ describe('.parse() general', () => {
   it('should not parse array type data', () => {
     expect(parser.parse(() => {})).toMatchObject({error: '#ERROR!', result: null});
   });
+
+  it('should parse scientific notation with whitespace', () => {
```
Comment (logansnow17) [inline test/integration/parsing/general.js:44]: Check these tests, and potentially suggest other ones

File: src/grammar-parser/grammar-parser.jison [hunk 4]
```diff
@@ -209,19 +210,28 @@ variableSequence
...
-      $$ = ($1 + '.' + $3) * 1;
+      $$ = ($1 + '.' + $3);
     }
   | DECIMAL NUMBER {
-      $$ = (0 + '.' + $2) * 1;
+      $$ = (0 + '.' + $2);
+    }
+;
+
+number
```
Comment (logansnow17) [inline src/grammar-parser/grammar-parser.jison:225]: These are the main changes

**Review by leb2 — APPROVED** [review]

**Pull request merged**